src/databricks/sql/client.py

```diff
@@ -297,6 +300,117 @@ def _check_not_closed(self):
         if not self.open:
             raise Error("Attempting operation on closed cursor")
 
+    def _handle_staging_operation(self, uploads_base_path: str):
+        """Fetch the HTTP request instruction from a staging ingestion command
+        and call the designated handler.
+        
+        Raise an exception if localFile is specified by the server but the localFile
+        is not descended from uploads_base_path.
+        """
+
+        if uploads_base_path is None:
+            raise Error(
+                "You must provide an uploads_base_path when initialising a connection to perform ingestion commands"
+            )
+        
+        row = self.active_result_set.fetchone()
+
+        if getattr(row, "localFile", None):
+            if os.path.commonpath([row.localFile, uploads_base_path]) != uploads_base_path:
+                raise Error("Local file operations are restricted to paths within the configured uploads_base_path")
```

Comment:
This may need to be more sophisticated. I'm considering what valid `uploads_base_path` could be provided where the longest common path shared by both files _is not_ identical to `uploads_base_path`. The idea here is that if the safe path is `/path/to/file` and the server specifies `/path/to/file/somewhere/on/disk` then the longest common path is `/path/to/file`.

Reply:
I think we should do that from a usability perspective before merge. the use case here is

User 1:
```
PUT /path/to/file/$userID1/2022-12-05 into $PSL_USER1

COPY INTO $deltaTable FROM $PSL_USER1
```

User 2:
```
PUT /path/to/file/$userID2/2022-12-05 into $PSL_USER2

COPY INTO $deltaTable2 FROM $PSL_USER2
```

We would like 2 user to use uploads_base_path and provide some sort of isolation among them.

WDYT ?

Reply:
Also, we should add some tests like this, we need to make sure the resolved path still start with uploads_base_path
```
PUT $uploads_base_path/../a.file into $PSL_USER2
```

Reply:
> We would like 2 user to use uploads_base_path and provide some sort of isolation among them.
> 
> WDYT ?

I don't think we have a clean way to limit user2 from accessing user1's `uploadsBasePath`. If `uploadsBasePath` is configured for a connection, then anyone with access to that connection can PUT files from that location. If isolation is required by the client then they should maintain a separate `Client()` for each user.

cc: @moderakh

Reply:
> Also, we should add some tests like this, we need to make sure the resolved path still start with uploads_base_path

@HSCandYH Done as part of 34a0362a81de1c5ceeeaca681d084fb2e9ed5249 👌